cpp/src/Slice/Grammar.y

```diff
@@ -1208,134 +1080,99 @@ class_extends
 extends
 // ----------------------------------------------------------------------
 : ICE_EXTENDS
-{
-}
 | ':'
-{
-}
-;
-
-// ----------------------------------------------------------------------
-class_exports
-// ----------------------------------------------------------------------
-: local_metadata class_export ';' class_exports
-{
-    StringListTokPtr metaData = StringListTokPtr::dynamicCast($1);
-    ContainedPtr contained = ContainedPtr::dynamicCast($2);
-    if(contained && !metaData->v.empty())
-    {
-        contained->setMetaData(metaData->v);
-    }
-}
-| error ';' class_exports
-{
-}
-| local_metadata class_export
-{
-    unit->error("`;' missing after definition");
-}
-| %empty
-{
-}
 ;
 
 // ----------------------------------------------------------------------
 data_member
 // ----------------------------------------------------------------------
-: tagged_type_id
```

Comment:
I split up 'type_id' and 'tagged_type_id' since both of these require a type and an identifier.
For return types, we don't have an identifier, so it's actually more useful to have a 'tagged_type' rule.

'tagged_type' applies to parameters, data members, and return values, since they can all have tags and types.
Whereas 'type_id' only applies to parameters and data members, but not return values since they don't always have identifiers.

Reply:
type_id means / meant type followed by identifier, and you're eliminating it with this PR?

Reply:
Yes, exactly. Something like 'bool hello' would be a type_id. It's not that isn't a useful grouping, it's just that grouping together tag+type is more useful than type+identifier. So type_id was eliminated and tagged_type was added.